src/thunderbots/software/util/visualizer_messenger/visualizer_messenger.cpp

```diff
@@ -11,20 +20,55 @@ namespace Util
         return vm;
     }
 
-    void VisualizerMessenger::initializePublisher(ros::NodeHandle node_handle)
+    void VisualizerMessenger::onConnection()
     {
-        if (this->publisher)
+        // The io_context is required for all I/O
+        boost::asio::io_service ioc{1};
+
+        // The acceptor receives incoming connections
+        // TODO: make IP and port a configurable constant somewhere?
+        auto const address = boost::asio::ip::address::from_string("127.0.0.1");
+        auto const port    = static_cast<unsigned short>(std::atoi("9091"));
+        tcp::acceptor acceptor{ioc, {address, port}};
+        for (;;)
         {
-            LOG(WARNING) << "Re-initializing visualizer messenger singleton publisher";
-        }
+            // This will receive the new connection
+            tcp::socket socket{ioc};
+
+            // Block until we get a connection
+            acceptor.accept(socket);
+
+            // Move the connection to a websocket stream
+            websocket::stream<tcp::socket> websocket(std::move(socket));
+
+            // Accept the websocket handshake
+            websocket.accept();
+
+            // Set the websocket to talk in binary
+            websocket.binary(true);
+
+            LOG(INFO) << "Connection received." << std::endl;
+
+            // Lock the current list of sockets
+            ws_mutex.lock();
```

Comment:
could we change `ws_mutex` to `websocket_list_mutex`? It's a small thing, but it really improves readability (particularly the first time over).

Reply:
Changed